### PR TITLE
Add process revocations functionality

### PR DESCRIPTION
**Pull request by aydarng**

### CODE REVIEW

File: src/verifier/core/basing.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def __iter__(self):
 CRED_CRYPT_INVALID = "Credential cryptographically invalid"
 CRED_CRYPT_VALID = "Credential cryptographically valid"
 CRED_AGE_OFF = "Credential presentation has aged off"
+AUTH_REVOKED = "Credential presentation has aged off"
```
Comment (2byrds) [inline src/verifier/core/basing.py:32]: looks like this was a copy paste description, needs updated

File: src/verifier/core/authorizing.py [hunk 2]
```diff
@@ -131,7 +130,9 @@ def processPresentations(self):
             cred_state = None
             if state.state == AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth*2):
                 self.vdb.iss.rem(keys=(aid,))
-            elif state.state != AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth):
+            # We keep revoked credentials in the DB because their auth should never expire and the state
+            # must always be AUTH_REVOKED to avoid logging in again with the older version of the credential
+            elif state.state != AUTH_REVOKED and state.state != AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth):
```
Comment (2byrds) [inline src/verifier/core/authorizing.py:135]: Do you think we need an additional elif for AUTH_SUCCESS, where we process_revocation again, just like we did in the 'verifying' stage.... the reason is that we will eventually have a watcher that is monitoring the credential status regularly. If the AUTH_SUCCESS occurred and then later the issuer revokes the credential, we need to update our status to AUTH_REVOKED. I know we don't have the watcher now.... and now that i'm thinking about it, perhaps if the watcher sees an update to revoked... then we'll process that automatically as a new presentation... hm, yea i think maybe that is better but what do you think @aydarng . If its a new presentation, can you just test a DIFFERERNT user (such as the issuer) presenting the revoked credential and verify that the issuee account is removed.

**Review by 2byrds — CHANGES_REQUESTED** [review]
Looking great. One small change and one watcher scenario (or other presenter scenario) to consider so that we don't rely on the user to send us their revoked cred.

File: src/verifier/core/authorizing.py (hunk 2 above)
Comment (aydarng) [reply src/verifier/core/authorizing.py:135]: Sure, I'll test that.

Comment (2byrds) [reply src/verifier/core/authorizing.py:135]: Great, let me know if its successful i'll approve it.

**Review by 2byrds — APPROVED** [review]
LGTM

**Pull request merged**